ecurve.js

```diff
@@ -163,15 +164,16 @@ function privateSub (d, tweak) {
   return dt
 }
 
-function sign (hash, x) {
+function sign (hash, x, addData) {
```

Comment:
In retrospect,  I think this is too easy to accidentally do

Reply:
How so?

Perhaps we should make a new function called signWithEntropy???

* sign -> __sign
* then sign calls sign with addData not used
* signWithEntropy is allowed to pass the addData

Thoughts?

Reply:
That is better

Reply:
@dcousens My NAN skills are lacking, it would probably be faster if you could just add the proper methods. I will modify the tests next.

Reply:
nvm, just copy pasted... not the best thing in the world, but at least tests are passing.

If you can tease it out into a separate function and call it from two NAN_METHODs then that would probably be better.